src/main/scala/chisel3/util/BitPat.scala

```diff
@@ -142,6 +143,57 @@ package experimental {
       bs.getWidth
       bs
     }
+
+    /**
+      * Construct a [[BitSet]] matching a range of value
+      *
+      * @param start The smallest matching value
+      * @param length The length of the matching range
+      * @param width The width of the constructed [[BitSet]]. If not given, the returned [[BitSet]] have the width of the maximum possible matching value.
+      * @return A [[BitSet]] matching exactly all inputs in range [start, start + length)
+      */
+    def fromRange(
+      start:  BigInt,
+      length: BigInt,
+      width:  Int = -1
+    ): BitSet = {
+      require(length > 0, "Cannot construct a empty BitSetRange")
+      val max_known_length = (start + length - 1).bitLength
```

Comment:
sorry for my nitpick
```suggestion
      val maxKnownLength = (start + length - 1).bitLength
```

Reply:
Fixed in c16799fd